core/src/test/scala/integration/kafka/api/ConsumerWithLegacyMessageFormatIntegrationTest.scala

```diff
@@ -80,12 +114,12 @@ class ConsumerWithLegacyMessageFormatIntegrationTest extends AbstractConsumerTes
     val timestampTopic2P0 = timestampOffsets.get(new TopicPartition(topic2, 0))
     assertEquals(40, timestampTopic2P0.offset)
     assertEquals(40, timestampTopic2P0.timestamp)
-    assertEquals(Optional.of(0), timestampTopic2P0.leaderEpoch)
+    assertEquals(Optional.empty, timestampTopic2P0.leaderEpoch)
```

Comment:
The commit https://github.com/apache/kafka/commit/8cdf1abb0b89f1f9f85fb1acb4cc8b83f527d16a included KRaft support for `ConsumerWithLegacyMessageFormatIntegrationTest`. However, it didn't really set topic to use v0 format. The KRaft used inter broker protocol version 3.0-IV1, so the message format version is ignored. That's why we could get leader epoch for topic2.

https://github.com/apache/kafka/blob/c455c10405a4a201912e35a52ca1e6aef36e5af9/core/src/main/scala/kafka/server/ConfigHandler.scala#L173-L177

Reply:
yes, we don't set the leader epoch for the old message format.